### PR TITLE
GCF file split: concepts

### PR DESCRIPTION
**Pull request by ace-n**
cc @saturnism

N.B: I ran `mvn dependency:analyze -DignoreNonCompile` to remove unused dependencies, but **this does not detect unused `test` dependencies.**

(If someone could point me to a command that identifies those, that would be appreciated. 🙂)

### CODE REVIEW

**Comment by kurtisvg** [comment]
@ace-n `-DignoreNonCompile` ignores test dependencies. Cant you just not use that flag?

File: functions/concepts/env-vars/pom.xml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava-testlib</artifactId>
+      <version>29.0-jre</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- Required for Function primitives -->
+    <dependency>
```
Comment (kurtisvg) [inline functions/concepts/env-vars/pom.xml:95]: nit: Please move the compile time dependencies to the top of the list

File: functions/concepts/execution-count/pom.xml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava-testlib</artifactId>
+      <version>29.0-jre</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- Required for Function primitives -->
+    <dependency>
+      <groupId>com.google.cloud.functions</groupId>
+      <artifactId>functions-framework-api</artifactId>
```
Comment (kurtisvg) [inline functions/concepts/execution-count/pom.xml:103]: nit: Compile time dependencies at the top

**Comment by ace-n** [comment]
Kurtis - I'm looking for something that _doesn't_ ignore test dependencies. 🙂

File: functions/concepts/lazy-fields/pom.xml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava-testlib</artifactId>
+      <version>29.0-jre</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- Required for Function primitives -->
+    <dependency>
+      <groupId>com.google.cloud.functions</groupId>
+      <artifactId>functions-framework-api</artifactId>
```
Comment (kurtisvg) [inline functions/concepts/lazy-fields/pom.xml:89]: nit: to the top!

File: functions/concepts/retry-pub-sub/pom.xml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava-testlib</artifactId>
+      <version>29.0-jre</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- Required for Function primitives -->
+    <dependency>
+      <groupId>com.google.cloud.functions</groupId>
+      <artifactId>functions-framework-api</artifactId>
```
Comment (kurtisvg) [inline functions/concepts/retry-pub-sub/pom.xml:103]: nit: to the top!

File: functions/concepts/scopes/pom.xml [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava-testlib</artifactId>
+      <version>29.0-jre</version>
+      <scope>test</scope>
+    </dependency>
+
+    <!-- Required for Function primitives -->
+    <dependency>
+      <groupId>com.google.cloud.functions</groupId>
+      <artifactId>functions-framework-api</artifactId>
```
Comment (kurtisvg) [inline functions/concepts/scopes/pom.xml:89]: nit: to the top

**Comment by kurtisvg** [comment]
> Thanks Kurtis - but I'm looking for something that _doesn't_ ignore test dependencies. 🙂
> 
> (In any case, this is more a "nice-to-have" than anything.)

Right. I'm suggesting you _don't_ use the flag that tells it to ignore test dependencies.

From the [ignoreNonCompile](http://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html#ignoreNonCompile) doc: 

>Ignore Runtime/Provided/Test/System scopes for unused dependency analysis.

So presumably the default is to include "runtime/provided/test/system scopes".

**Comment by ace-n** [comment]
Ah - my bad, I assumed some sort of flag was required! 🙃

**Pull request merged**